FastTag.md

```diff
@@ -0,0 +1,225 @@
+}
+.fastnav-pages span:hover {
+  cursor: pointer;
+  background-color: #e6e6e6;
+}
+
+.fastnav-block {
+  color: #222;
+}
+```
+
+```space-lua
+fastnav = fastnav or {}
+
+-- Fetch all relevant page and tag data
+--     where p.page.startsWith("projects/")
+local function fetchTagAndPageData(dir)
+  return query[[
+  from p = index.tag "page"
+  where p.name.startsWith(dir)
+  order by p.name 
+  select {
+      name = p.name,
+      tags = p.tags
+    }
+  ]]
+end
+
+-- Convert skiplist into a lookup set
+local function buildSkipSet(skiplist)
+  local skip = {}
+  for _, name in ipairs(skiplist or {}) do
+    skip[name] = true
+  end
+  return skip
+end
+
+-- Analyse all pages: count tags, check for untagged pages
+local function analyseTags(data, skip)
+  local tagCounts = {}
+  local hasUntagged = false
+  for _, page in ipairs(data) do
+    local tags = page.tags
+    if not tags or #tags == 0 then
+      hasUntagged = true
+    else
+      for _, tag in ipairs(tags) do
+        if not skip[tag] then
+          tagCounts[tag] = (tagCounts[tag] or 0) + 1
+        end
+      end
+    end
+  end
+  return tagCounts, hasUntagged
+end
+
+-- Convert tag count map to sorted array
+local function sortedTagList(counts)
+  local tags = {}
+  for tag, count in pairs(counts) do
+    table.insert(tags, {name=tag, count=count})
+  end
+  table.sort(tags, function(a, b) return a.name < b.name end)
+  return tags
+end
+
+-- Generate HTML for all tags
+function fastnav.TagsHtml(data, skip)
+  local tagCounts, hasUntagged = analyseTags(data, skip)
+  local sortedTags = sortedTagList(tagCounts)
+  -- Generate the default tags
+  local html = {}
+  table.insert(html, fastnav.TagHtml("AllTags"))
+  if hasUntagged then
+    table.insert(html, fastnav.TagHtml("NoTags"))
+  end
+  -- Generate tag buttons associated with the pages
+  local b = 1
+  for _, tag in ipairs(sortedTags) do
+    table.insert(html, fastnav.TagHtml(tag.name, tag.count))
+  end
+  return html
+end
+
+-- Generate HTML for all pages
+function fastnav.PagesHtml(data, skip)
+  local html = {}
+  for _, page in ipairs(data) do
+    local tags = page.tags or {}
+    local skipPage = false
+    for _, tag in ipairs(tags) do
+      if skip[tag] then
+        skipPage = true
+        break
+      end
+    end
+    if not skipPage then
+      local tagList = (#tags == 0) and { "NoTags" } or tags
+      table.insert(html, fastnav.PageHtml(page.name, tagList))
+    end
+  end
+  return html
+end
+
+-- Individual page blocks
+function fastnav.PageHtml(name, tagList)
+  local tagClass = table.concat(tagList, " ")
+  local basename = string.match(name, "[^/]+$") or name
+  local js = "window.location='"..name.."';"
+  return '<span onclick="'..js.. '" class="'..tagClass..'">'..basename..'</span>'
+end
+
+-- Individual tag buttons
+function fastnav.TagHtml(name, count)
+  local js = ""
+  local all = "AllTags"
+  if name == all then
+    js = js .. "window.activeTags=[];"
```

Comment:
Can you use `[[` style strings perhaps with spacelua.interpolate as I suggested earlier?